src/Raven.Server/Documents/AbstractBackgroundWorkStorage.cs

```diff
@@ -183,25 +189,36 @@ public static bool HasPassed(BlittableJsonReaderObject metadata, DateTime curren
     
     protected abstract void ProcessDocument(DocumentsOperationContext context, Slice lowerId, string id, DateTime currentTime);
 
-    public int ProcessDocuments(DocumentsOperationContext context, Dictionary<Slice, List<(Slice LowerId, string Id)>> docsToProcess, DateTime currentTime)
+    public int ProcessDocuments(DocumentsOperationContext context, Queue<DocumentExpirationInfo> expired, DateTime currentTime)
```

Comment:
we should avoid using the word `expired` here because some of these could've not expired yet

Reply:
done